sensing/autoware_pointcloud_preprocessor/src/distortion_corrector/distortion_corrector.cpp

```diff
@@ -208,73 +213,81 @@ bool DistortionCorrector<T>::azimuthConversionExists(sensor_msgs::msg::PointClou
 
     // If the angle exceeds 180 degrees, it may cross the 0-degree axis,
     // which could disrupt the calculation of the formula.
     if (
-      abs(*next_it_azimuth - *it_azimuth) >= autoware::universe_utils::pi ||
-      abs(next_cartesian_rad - current_cartesian_rad) >= autoware::universe_utils::pi) {
-      RCLCPP_WARN(
-        node_->get_logger(),
-        "Angle between two points exceeds 180 degrees. Iterate to next point ...");
+      abs(*next_it_azimuth - *it_azimuth) == 0 ||
+      abs(next_cartesian_rad - current_cartesian_rad) == 0) {
+      RCLCPP_DEBUG(
+        node_->get_logger(), "Angle between two points is 0 degrees. Iterate to next point ...");
       continue;
     }
 
-    float b = (*next_it_azimuth - *it_azimuth) / (next_cartesian_rad - current_cartesian_rad);
-    float a = *it_azimuth - b * current_cartesian_rad;
-
-    // Check if 'b' can be adjusted to 1 or -1
-    if (std::abs(b - 1.0f) <= threshold_b_) {
-      b_ = 1.0f;
-    } else if (std::abs(b + 1.0f) <= threshold_b_) {
-      b_ = -1.0f;
+    // restrict the angle difference between [-180, 180] (degrees)
+    float azimuth_diff = abs(*next_it_azimuth - *it_azimuth) > autoware::universe_utils::pi
+                           ? abs(*next_it_azimuth - *it_azimuth) - 2 * autoware::universe_utils::pi
+                           : *next_it_azimuth - *it_azimuth;
+    float cartesian_rad_diff =
+      abs(next_cartesian_rad - current_cartesian_rad) > autoware::universe_utils::pi
+        ? abs(next_cartesian_rad - current_cartesian_rad) - 2 * autoware::universe_utils::pi
+        : next_cartesian_rad - current_cartesian_rad;
+
+    float sign = azimuth_diff / cartesian_rad_diff;
+
+    // Check if 'sign' can be adjusted to 1 or -1
+    if (std::abs(sign - 1.0f) <= angle_conversion.sign_threshold) {
+      angle_conversion.sign = 1.0f;
+    } else if (std::abs(sign + 1.0f) <= angle_conversion.sign_threshold) {
+      angle_conversion.sign = -1.0f;
     } else {
-      RCLCPP_WARN(
-        node_->get_logger(),
-        "Angle between two points exceeds 180 degrees. Iterate to next point ...");
-      std::cout << "b: " << b << "should be close to 1 or -1" << std::endl;
+      RCLCPP_DEBUG(
+        node_->get_logger(), "Value of sign is not close to 1 or -1. Iterate to next point ...");
       continue;
     }
 
-    // Check if 'a' can be adjusted to a multiple of π/2
-    int multiple_of_90_degrees = std::round(a / (autoware::universe_utils::pi / 2));
-    if (std::abs(a - multiple_of_90_degrees * (autoware::universe_utils::pi / 2)) > threshold_a_) {
+    float offset_rad = *it_azimuth - sign * current_cartesian_rad;
+    // Check if 'offset_rad' can be adjusted to offset_rad multiple of π/2
+    int multiple_of_90_degrees = std::round(offset_rad / (autoware::universe_utils::pi / 2));
+    if (
+      std::abs(offset_rad - multiple_of_90_degrees * (autoware::universe_utils::pi / 2)) >
+      angle_conversion.offset_rad_threshold) {
+      RCLCPP_DEBUG(
```

Comment:
Use throttle here

Reply:
all of the logging are fixed in https://github.com/autowarefoundation/autoware.universe/pull/8380/commits/03354c85ca8ec274c2360466078429d7d782d2e9